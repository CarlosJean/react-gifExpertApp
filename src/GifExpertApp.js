import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";
import 'animate.css';

export const GifExpertApp = ( { defaultCategories = [] }) => {
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories}/>
            <hr/>
            <ol>
                {
                    categories.map(category => {
                        return <GifGrid category = {category} key = {category}/>
                    })
                }
            </ol>
        </>
    );

}
