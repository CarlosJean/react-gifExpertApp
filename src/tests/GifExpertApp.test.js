import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe('Pruebas al componente GifExpertApp', () => {
    test('Debe coincidir con el snapshot', () => {
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });
    
    test('Debe coincidir con la cantidad de grids.', () => {

        const categories =  ['One Punch', 'Dragon Ball'];

        const wrapper = shallow(<GifExpertApp defaultCategories = {categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});