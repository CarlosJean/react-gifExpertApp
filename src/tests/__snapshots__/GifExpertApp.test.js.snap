// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente GifExpertApp Debe coincidir con el snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas al componente GifExpertApp Debe coincidir con la cantidad de grids. 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
